src/components/ProgressBar/ProgressBar.tsx

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import { useUID } from 'react-uid';
+import styles from './ProgressBar.module.css';
+import Text from '../Text';
+
+export type Props = {
+  /**
+   * CSS class names that can be appended to the component.
+   */
+  className?: string;
+  /**
+   * Label associated with naming the progress bar.
+   */
+  label: string;
+  /**
+   * Max value to be represented by the progress bar.
+   */
+  max: number;
+  /**
+   * The amount of segments to be represented by the progress bar.
+   * Recommended max number of segments is 10.
+   */
+  segmentCount: number;
+  /**
+   * The value that each segment represents.
+   */
+  segmentValue: number;
+  /**
+   * An optional unit of measurement to describe the progress.
+   */
+  unit?: string;
+};
+
+/**
+ * BETA: This component is still a work in progress and is subject to change.
+ *
+ * `import {ProgressBar} from "@chanzuckerberg/eds";`
+ *
+ * A progress bar component that indicating the completion progress of a task.
+ */
+export const ProgressBar = ({
+  className,
+  label,
+  segmentCount,
+  segmentValue,
+  max,
+  unit,
+  ...other
+}: Props) => {
+  if (process.env.NODE_ENV !== 'production' && segmentCount > 10) {
+    console.warn('Progress bar segment count should be capped at 10');
+  }
```

Comment:
I'm not sure how I feel about this warning. If it were a hard requirement, I would suggest we use types (something ugly like `segmentCount = 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10;`), but if it's a recommendation, we could end up with warnings clogging up the terminal when the component is working as designed. I think it's enough to have this in the docstring for the component and in design documentation in zeroheight

Reply:
Ah agree that typing as enum numbers would be better here

Reply:
Ugh yikes. This SO question has another idea but that might also be an issue here: https://stackoverflow.com/questions/39494689/is-it-possible-to-restrict-number-to-a-certain-range

I'm also fine with just putting it in the documentation. I just don't want to keep adding to the browser console mess LP already has if we can help it.

Reply:
I am OK with us removing this code constraint as a warning. When designs are produced using this component, they should follow the recommendations in zeroheight and whatever we document. But if someone does use more segments, the implementation will handle it to a certain degree (if it breaks the rules, it's a conversation that the designers and engineers need to talk thru for that page)

Handling > 10 but documenting "only ten or less, please" follows the [Robustness Principle](https://en.wikipedia.org/wiki/Robustness_principle) w/o making too much noise...

Reply:
Sounds good, will remove warning and just keep as type number